File: Alay/Sliding-Window/3208.cpp
class Solution {
    public int numberOfAlternatingGroups(int[] colors, int k) {

        int answer = 0;
        int count = 0;
        int length = colors.length;

        for (int i = 1, start = 0; i < 2 * length - 1; i++) {
            if (i == length - 1) {
                count = answer;
            }
            if (colors[i % length] == 1 && colors[(i - 1) % length] == 0) {
                if (i - start + 1 >= k) {
                    ++answer;
                }
            }
            else if (colors[i % length] == 0 && colors[(i - 1) % length] == 1) {
                if (i - start + 1 >= k) {
                    ++answer;
                }
            }
            else {
                start = i;
            }
        }

        return answer - count;
    }
}

File: [Manya]/Leetcode/Sliding-Window/3208.java
import java.util.Scanner;

class Solution {
    public int numberOfAlternatingGroups(int[] colors, int k) {
        int n = colors.length;
        int count = 0;
        int validLength = 0;

        for (int i = 0; i < n * 2; i++) {
            if (colors[i % n] != colors[(i - 1 + n) % n]) {
                validLength++;
            } else {
                validLength = 1;
            }

            if (validLength >= k && i >= n) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter number of elements in the colors array: ");
        int n = scanner.nextInt();

        int[] colors = new int[n];
        System.out.println("Enter the elements of the colors array:");
        for (int i = 0; i < n; i++) {
            colors[i] = scanner.nextInt();
        }

        System.out.print("Enter the value of k: ");
        int k = scanner.nextInt();

        Solution solution = new Solution();
        int result = solution.numberOfAlternatingGroups(colors, k);

        System.out.println("Number of alternating groups of length at least k: " + result);

        scanner.close();
    }
}
